controllers/trainer_controller.go

```diff
@@ -1,27 +1,73 @@
 package controllers
 
 import (
+	"fmt"
 	"net/http"
 
 	"trainder-api/models"
 	"trainder-api/responses"
+
 	"trainder-api/utils/tokens"
 
 	"github.com/gin-gonic/gin"
 )
 
+type FilterTrainerInput struct {
+	Speciality []string `json:"Speciality" binding:"required"`
+	// Rating     float32 `json:"Rating" binding:"required"`
+	// Fee        float32 `json:"Fee" binding:"required"`
+}
 type TrainerInput struct {
 	Speciality     []string `json:"speciality" `
-	Rating         float64  `json:"raiting "`
-	Fee            float64  `json:"fee `
-	TraineeCount   int32    `json:"traineeCount `
-	CertificateUrl string   `json:"certificateUrl`
+	Rating         float64  `json:"rating "`
+	Fee            float64  `json:"fee" `
+	TraineeCount   int32    `json:"traineeCount" `
+	CertificateUrl string   `json:"certificateUrl"`
 }
 
 type GetTrainerInput struct {
 	Username string `json:"username" binding:"required"`
 }
 
+// FilterTrainer godoc
+//
+//	@Summary		FilterTrainer base on filter input
+//	@Description	FilterTrainer base on filter input
+//	@Tags			user
+//	@Accept			json
+//	@Produce		json
+//	@Param			json_in_ginContext	body		FilterTrainerInput	true	"put FilterTrainerInput input json and pass to  gin.Context"
+//	@Success		200					{object}	responses.FilterTrainerResponses
+//
+//	@Router			/protected/filter-trainer [get]
+func FilterTrainer() gin.HandlerFunc {
```

Comment:
This function should also accept the `limit` parameter in which when the parameter is specified the api should only respond with min(len(trainerThatExistsWithFilter), limit) trainer(s).